addons/stock/models/product.py

```diff
@@ -365,8 +365,9 @@ def _compute_nbr_reordering_rules(self):
     def onchange_tracking(self):
         products = self.filtered(lambda self: self.tracking and self.tracking != 'none')
         if products:
-            unassigned_quants = self.env['stock.quant'].search_count([('product_id', 'in', products.ids), ('lot_id', '=', False), ('location_id.usage','=', 'internal')])
-            if unassigned_quants:
+            unassigned_quants = self.env['stock.quant'].search([('product_id', 'in', products.ids), ('lot_id', '=', False), ('location_id.usage','=', 'internal')])
```

Comment:
Use read_group instead, You will compute the sum in SQL, so it's much quicker. You can find a good example here : https://github.com/odoo/odoo/blob/a52fbb1278cfeab9c53cd1f27aa98306f4974722/addons/mrp/report/report_stock_forecasted.py#L37